pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -197,19 +205,19 @@ func (rq *Controller) enqueueAll() {
 }
 
 // obj could be an *v1.ResourceQuota, or a DeletionFinalStateUnknown marker item.
-func (rq *Controller) enqueueResourceQuota(obj interface{}) {
+func (rq *Controller) enqueueResourceQuota(ctx context.Context, obj interface{}) {
 	key, err := controller.KeyFunc(obj)
 	if err != nil {
-		klog.Errorf("Couldn't get key for object %+v: %v", obj, err)
+		klog.FromContext(ctx).Error(err, "Couldn't get key", "obj", spew.Sdump(obj))
```

Comment:
Dumping the entire object with a format that won't match the expected output from the logger (e.g. JSON) doesn't strike me as a super useful thing for someone reading the logs in the future, but I see that it's mostly just carrying forward previous behavior.

Reply:
I wouldn't pull in `spew` for this one usage, in particular when considering that this line should never get hit in practice. You can just pass `obj` and let the logging backend figure out how to format it.